### ISSUE TITLE
Async initialize methods on shapes are not awaited

### ISSUE THREAD

**Issue by paynecrl97** [OrchardCMS/OrchardCore#1082]
In a display driver, the following example would end up with the `model.List` property being null because the initialize method is not awaited:

    public override IDisplayResult Display(MyPart part)
    {
        return Shape<ViewVM>("MyPart_View", async model =>
        {
            model.List = await _myApi.DoSomethingAsync();
        }).Location("Detail", "Content:1");
    }

Offending line: https://github.com/OrchardCMS/OrchardCore/blob/master/src/OrchardCore/OrchardCore.DisplayManagement/Handlers/DisplayDriverBase.cs#L39

**Comment by sebastienros** [comment]
It actually shows an issue with the code, it's missing a Func<Task<Model>> overload, see http://thebillwagner.com/blog/Item/2016-05-18-DoasynclambdasreturnTasks

**Comment by paynecrl97** [comment]
I've actually just realised that this does not fix the issue I was seeing. I'll update the bug issue I raised with more details.

**Comment by paynecrl97** [comment]
I've just realised that the PR I raised does not fix the issue I was seeing.

What is happening is the first request that executes the driver seems to succeed, but all subsequent requests fail. Simply restarting IIS (i.e. no code changes) is enough to make the driver work again for a single request, and then it'll revert to failing.

I'm assuming this is a failure related to caching or memoization, but will look further in to the cause and post back here with my findings.
